lib/iris/_concatenate.py

```diff
@@ -431,9 +431,8 @@ def match(self, other, error_on_mismatch):
 
         # Check if either cube is anonymous.
         if self.anonymous or other.anonymous:
-            msg = ('Dimensions differ: one or both cubes have anonymous '
-                   'dimensions')
-            msgs.append(msg)
+            msg = ('One or both cubes have anonymous dimensions')
```

Comment:
I think I used a warning as I was concerned about someone trying to concatenate a cube that wasn't fully defined. However, I realise this is a risk with any concatenation, regardless of whether there is an anonymous dimension.

I will put up another pull request that removes this warning.